src/Datastore/Key.php

```diff
@@ -0,0 +1,379 @@
+<?php
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Datastore;
+
+use InvalidArgumentException;
+use JsonSerializable;
+
+/**
+ * Represents a Datastore Key.
+ *
+ * Keys are unique identifiers for entities.
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\ServiceBuilder;
+ *
+ * $cloud = new ServiceBuilder();
+ * $datastore = $cloud->datastore();
+ *
+ * $key = $datastore->key('Person', 'Bob');
+ * ```
+ *
+ * @see https://cloud.google.com/datastore/reference/rest/v1beta3/Key Key
```

Comment:
The referenced documentation marks itself as deprecated - I think it'd make sense to reference `v1` in the doc-comments as well.